### PR TITLE
when spawning a pty, be sure to provide & escape conhost's path

### LINKED ISSUE #4061: Terminal failed to start if "C:\Program.exe" exists
# Environment

```none
Windows build number: Microsoft Windows NT 10.0.18363.0

Windows Terminal version: Version: 0.7.3451.0

A program named "C:\Program.exe" by myself
```
Here is the source code of C:\Program.exe
```c
#include <windows.h>
int WinMain(
    HINSTANCE hInstance,
    HINSTANCE hPrevInstance,
    LPSTR lpCmdLine,
    int nShowCmd)
{
    MessageBox(NULL, "C:\\Program.exe Executed!", "Bug", MB_OK);
    return 0;
}
```
my profile
```json
        {
            "guid": "{58ad8b0c-3ef8-5f4d-bc6f-13e4c00f2530}",
            "hidden": false,
            "name": "Debian",
            "fontFace": "等距更纱黑体 SC",
            "source": "Windows.Terminal.Wsl"
        },
```
# Steps to reproduce

rename the exe to "Program.exe" and place it at C:\ then run Windows Terminal

# Expected behavior

Windows Terminal starts normally

# Actual behavior

C:\Program.exe is executed. And after it exits, Windows Terminal exit with an error

捕获
捕获
捕获3

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
Fixes #4061.

I've tested this in multiple build and deployment configurations and placed a console application at `C:\Program.exe`

## before

Terminal spawns Program.exe instead of conhost.exe, and then tries to attach CMD to it.

image

_Program.exe_ prints out "UNICODE. ^Z TO EXIT".

## after

image

Everything is good.

### CODE REVIEW

File: src/winconpty/winconpty.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ HRESULT _CreatePseudoConsole(const HANDLE hToken,
     RETURN_IF_WIN32_BOOL_FALSE(CreatePipe(signalPipeConhostSide.addressof(), signalPipeOurSide.addressof(), &sa, 0));
     RETURN_IF_WIN32_BOOL_FALSE(SetHandleInformation(signalPipeConhostSide.get(), HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT));
 
-    const wchar_t* pwszFormat = L"%s --headless %s--width %hu --height %hu --signal 0x%x --server 0x%x";
+    const wchar_t* pwszFormat = L"\"%s\" --headless %s--width %hu --height %hu --signal 0x%x --server 0x%x";
```
Comment (zadjii-msft) [inline src/winconpty/winconpty.cpp]: maybe add a comment referring back to this issue? I feel like this is something we'd forget

**Comment by zadjii-msft** [comment]
@msftbot make sure @miniksa signs off

File: src/winconpty/winconpty.cpp (hunk 1 above)
Comment (miniksa) [reply src/winconpty/winconpty.cpp]: did it

**Comment by miniksa** [comment]
> Huh, there's no reason x86 should have failed here.
> 
> /azp run

This is the same thing that happened to me yesterday or the day before. When I merged master INTO my branch, everything was then fine. So I just clicked the button to do that.

**Comment by DHowett-MSFT** [comment]
(I'm having a look locally.)

**Comment by DHowett-MSFT** [comment]
So, it's failing with `0xC0000096 STATUS_PRIVILEGED_INSTRUCTION` which makes very little sense.

**Comment by DHowett-MSFT** [comment]
Okay, so I'm force-merging this.

_I pulled the test binaries from the test machine. These are the same ones that fail in CI._

```
Summary: Total=34950, Passed=34950, Failed=0, Blocked=0, Not Run=0, Skipped=0
...

```

wait what, the tests passed when i force pushed it?

```

**Comment by DHowett-MSFT** [comment]
the same failure is hitting intermittently on amster.

**Pull request merged**